### ISSUE TITLE
Failure to generate Borderline Instance

### ISSUE THREAD

**Issue by fankaisheng** [hamidkarimi/DeepDIG#2]
I trained AE1_s_t and AE2_s_t_s according to the settings of the paper and the reference code, and successfully to get: 
1). AE1_s_t generates " \hat_{Xt}" based on real samples and f( \hat_{Xt}) = label_t;
2). AE2_s_t_s generates " \hat_{Xs}" based on real samples and f( \hat_{Xs}) = label_s.
But I failed in finding the Borderline Instance, there is no x_m such that f_s(x_m) - f_t(x_m) < threshold_beta after I made sure that every pair \hat_{xt} and \hat_{xs} is correct.

In other words x_m is predicted by the model to end up in a category other than s and t every time, is this because my AE training is not good enough?

Using MNIST as an example, I would appreciate it if you could upload the adv_data/adv_adv_data generated by component 1 and component 2 of the paper, I may be able to successfully find the Borderline Instance.

**Comment by fankaisheng** [comment]
I have successfully solved the problem of not being able to find the Borderline Instance.

The reason is that component 2, 'AE2_s_t_s' overfitting, causes the mode collapse and all the samples generated by the second AE are visualized to come out the same.

I was able to find Borderline Instance using real samples and adversarial samples generated by AE1, where the threshold beta=0.001. This makes designing the second component feel non-essential.

**Comment by hamidkarimi** [comment]
> I have successfully solved the problem of not being able to find the Borderline Instance.
> 
> The reason is that component 2, 'AE2_s_t_s' overfitting, causes the mode collapse and all the samples generated by the second AE are visualized to come out the same.
> 
> I was able to find Borderline Instance using real samples and adversarial samples generated by AE1, where the threshold beta=0.001. This makes designing the second component feel non-essential.

I am glad you figure it out. Perhaps in some cases component 2 may be non-essential, but I designed to be generic and applied to different data.
